pkg/component/delete/delete.go

```diff
@@ -74,3 +86,105 @@ func references(list []unstructured.Unstructured, ownerRef metav1.OwnerReference
 	}
 	return false
 }
+
+// ListResourcesToDeleteFromDevfile parses all the devfile components and returns a list of resources that are present on the cluster and can be deleted
+func (do DeleteComponentClient) ListResourcesToDeleteFromDevfile(devfileObj parser.DevfileObj, appName string) (isInnerLoopDeployed bool, resources []unstructured.Unstructured, err error) {
+	// Inner Loop
+	// Fetch the deployment of the devfile component
+	componentName := devfileObj.GetMetadataName()
+	var deploymentName string
+	deploymentName, err = util.NamespaceKubernetesObject(componentName, appName)
+	if err != nil {
+		return isInnerLoopDeployed, nil, fmt.Errorf("Failed to get the resource %q name for component %q; cause: %w", kclient.DeploymentKind, deploymentName, err)
+	}
+	deployment, err := do.kubeClient.GetDeploymentByName(deploymentName)
+	if err != nil && !kerrors.IsNotFound(err) {
+		return isInnerLoopDeployed, nil, err
+	}
+
+	// if the deployment is found on the cluster,
+	// then convert it to unstructured.Unstructured object so that it can be appended to resources;
+	// else if it is not found, then continue to outer loop
+	if err == nil {
+		isInnerLoopDeployed = true
+		var unstructuredDeploy unstructured.Unstructured
+		unstructuredDeploy, err = kclient.ConvertK8sResourceToUnstructured(deployment)
+		if err != nil {
+			return isInnerLoopDeployed, nil, fmt.Errorf("Failed to parse the resource %q: %q; cause: %w", kclient.DeploymentKind, deploymentName, err)
+		}
+		resources = append(resources, unstructuredDeploy)
+	}
```

Comment:
You could write a sub-function for this block, to be able to return ASAP in particular cases, for example the case where `IsNotFound(err) is true`.

The `err == nil` is difficult to interpret here, it should be better to have `deployment != nil` instead.

Reply:
I cannot check `deployment != nil` because it's never the case, I have settled for `deployment.Name != ""` instead.
I tried moving the block to a sub-function, but that made things worse with `isInnerLoopDeployed` involved, so I've kept things as they are.